src/ShopArticleReviewBundle/objects/WebModules/MTPkgShopArticleReviewCore/MTPkgShopArticleReviewCore.class.php

```diff
@@ -734,18 +732,23 @@ protected function GetArticleToReview()
 
     protected function GetReviewWriteData()
     {
-        $oGlobal = TGlobal::instance();
-        $aUserData = $oGlobal->GetuserData(TdbShopArticleReview::INPUT_BASE_NAME);
-        $aUserData['author_name'] = $this->GetAuthorName($aUserData['author_name']);
+        $inputData = $this->getInputFilterUtil()->getFilteredInput(TdbShopArticleReview::INPUT_BASE_NAME);
+
+        $formData['author_name'] = $this->GetAuthorName($inputData['author_name']);
+        $formData['rating'] = $inputData['rating'] ?? '5';
+        $formData['title'] = $inputData['title'] ?? '';
+        $formData['comment'] = $inputData['comment'] ?? '';
+        $formData['author_email'] = $inputData['author_email'] ?? '';
```

Comment:
I wonder what happens if  not logged-in but this email is already taken..

I don't see a check - for example in \MTPkgShopArticleReviewCore::ValidateWriteReviewData().

Reply:
Currently in the demo shop the review option is only usable once logged in.